scene/gui/rich_text_label.cpp

```diff
@@ -1100,6 +1117,23 @@ int RichTextLabel::_draw_line(ItemFrame *p_frame, int p_line, const Vector2 &p_o
 				}
 			}
 
+			Vector2 gl_off_2 = char_xform.columns[2];
+			Vector2i gl_off_2i = gl_off_2;
+
+			/*
+			// I have the feeling this is the actual correct math, but results look practically the same
+			Transform2D char_final_xform = Transform2D();
+			char_final_xform.columns[2] -= gl_off_2;
+			char_final_xform = char_xform * char_final_xform;
+			char_final_xform.columns[2] += Vector2(gl_off_2.x-gl_off_2i.x, gl_off_2.y-gl_off_2i.y);
+			*/
+
+			Transform2D char_final_xform = Transform2D();
+			char_final_xform.columns[2] -= gl_off_2i;
+			char_final_xform = char_xform * char_final_xform;
+
+			draw_set_transform_matrix(char_final_xform);
```

Comment:
This part should have performance tested with large number of glyphs (adding draw commands might have significant impact).

```suggestion
			if (char_final_xform != Transform2D()) {
				draw_set_transform_matrix(char_final_xform);
			}
```

Reply:
this hasn't been adressed yet, I think we can't directly compare to an identity transform, and instead we should keep the last pushed transform and compare against that